client/components/TitleEditor/TitleEditor.tsx

```diff
@@ -0,0 +1,185 @@
+import classNames from 'classnames';
+import { useBeforeUnload } from 'react-use';
+import React, { ClipboardEvent, useCallback, useEffect, useRef, useState } from 'react';
+
+require('./titleEditor.scss');
+
+type Props = {
+	isReadOnly?: boolean;
+	initialValue?: string;
+	onChange?: (value: string) => void;
```

Comment:
Thoughts on doing some tag stripping within this component, so that `onChange` has this signature instead?

```ts
onChange: (update: Pick<Pub, 'title' | 'htmlTitle'>) => unknown;
```

This way the Pub page gets to know its updated `title` in real time, which might be good. We'll probably still want to do the normalization on the server, too, which is annoying. (But this is good: we're starting to see bits of annoying synchronization logic that Facets can abstract over).